### PR TITLE
path fixes to make api more consistent

### LINKED ISSUE #8: does not handle branch name with "/"
the plugin as currently written uses the folder name that needs to match the branch name but I have branch names with "/". when I create a worktree everything after the "/" gets cut off

### PR DESCRIPTION
**Pull request by polarmutex**
fixes #8

### CODE REVIEW

**Comment by polarmutex** [comment]
this makes all of the api use path name and not branch name. we were using the branch name in finding the worktree (forcing you to name the worktree path the same name as the branch). Now you only need the branch name on creation. in my limited tests this fixes branches that have special chars in the name (ex '/')

File: lua/git-worktree/init.lua [hunk 1]
```diff
@@ -68,7 +68,7 @@ local function has_worktree(path, cb)
     local found = false
     local job = Job:new({
         'git', 'worktree', 'list', on_stdout = function(_, data)
-            local start = string.find(data, string.format("[%s]", path), 1, true)
+            local start = string.find(data, string.format("%s/%s", git_worktree_root, path), 1, true)
```
Comment (ThePrimeagen) [inline lua/git-worktree/init.lua:71]: Lets use Path.path.sep instead of `/` for the path separator

File: lua/git-worktree/init.lua [hunk 2]
```diff
@@ -103,6 +103,8 @@ local function has_branch(path, cb)
     local found = false
     local job = Job:new({
         'git', 'branch', on_stdout = function(_, data)
+            -- remove  markere on current branch
+            data = data:gsub("*","")
```
Comment (ThePrimeagen) [inline lua/git-worktree/init.lua:107]: SNAP!

File: lua/telescope/_extensions/git_worktree.lua [hunk 3]
```diff
@@ -1,4 +1,4 @@
-local Job = require("plenary.job")
+local Path = require("plenary.path")
```
Comment (ThePrimeagen) [inline lua/telescope/_extensions/git_worktree.lua:1]: @polarmutex could you fix the indenting?

Everything else *i think* is ok

Comment (ThePrimeagen) [reply lua/telescope/_extensions/git_worktree.lua:1]: It looks like you are using tabs and we be using 4 spaces!

**Comment by ThePrimeagen** [comment]
I APPROVE

**Pull request merged**